wordpress-mobile/WordPress-Android
# Issue/11119 search in set parent page

**Fixes #**:  https://github.com/wordpress-mobile/WordPress-Android/issues/11119

**Description**:
This PR adds search capabilities to the Set Parent page. It is 95% complete. Following along with the
pattern in Site Pages, this implementation separates the search into a `PageParentSearchFragment`
and `PageParentSearchVieModel`, both of which have been added to the necc. dagger modules. A new layout file (`page_parent_fragment.xml`) was introduced for `PageParentFragment` to hold the FrameLayout for the search and page_parent_menu.xml was expanded to hold a new SearchView. The searching takes advantage of LiveData, ViewModels and observability to provide the user with a clean UI experience. 

While this PR is 95% complete, there are still a few lingering questions that may require action. 
1 - Does the text need to be adjusted when showing the suggestion or empty results? Right now that 
text is shared as ... 
    <string name="pages_empty_search_result">No pages matching your search</string>
    <string name="pages_search_suggestion">Search pages</string>
If changes are necessary, two new string would be required + any language translations required.

2 - As of this PR submittal, the search is locally performed on the parent page options during page load. It is unclear if the search needs to be performed against the database each and every time. If so, then the search logic will need to be updated to reflect that.

**Testing instructions:**  
Test starting prompt is shown 
- Navigate to Site pages
- Tap on a action menu (3 vertical dots) for a page and Select "Set Parent"
- Tap on the search icon 
- "Search hint" is available on the toolbar
- "Search pages" is visible in the page_parent_fragment

Test search text found
- Navigate to Site pages
- Tap on a action menu for a page and Select "Set Parent"
- Tap on the search icon 
- Enter search text (jot dowan a title snippet from the pages view & search for that)
- A list of possible parent pages is shown

Test search text found
- Navigate to Site pages
- Tap on a action menu for a page and Select "Set Parent"
- Tap on the search icon 
- Enter jibberish as the search text
- "No pages matching your search" is shown

![Screenshot_1581281732](https://user-images.githubusercontent.com/506707/74111040-6c597600-4b5f-11ea-9ee7-0d1477fe3a67.png)
![Screenshot_1581281746](https://user-images.githubusercontent.com/506707/74111043-71b6c080-4b5f-11ea-82f5-f51feead04bd.png)
![Screenshot_1581281755](https://user-images.githubusercontent.com/506707/74111049-77aca180-4b5f-11ea-83f7-868fb987ecac.png)
![Screenshot_1581281775](https://user-images.githubusercontent.com/506707/74111055-7c715580-4b5f-11ea-9c9d-39320614d8a4.png)

![search-set-page-parent](https://user-images.githubusercontent.com/506707/74111058-84c99080-4b5f-11ea-9db0-82750b884ddd.gif)


PR submission checklist:
